cpp/include/cudf/detail/copy_if_else.cuh

```diff
@@ -93,8 +92,7 @@ void copy_if_else_kernel(  column_device_view const lhs,
 
       // next grid
       w_cur += warps_per_grid;
-      index += blockDim.x * gridDim.x;
-      w_active_mask = __ballot_sync(w_active_mask, w_cur <= w_end);      
+      index += blockDim.x * gridDim.x;      
```

Comment:
This still isn't right. You want 
```
index += blockDim.x * gridDim.x / warp_size.

Reply:
Actually no, you want to use the already computed number of warps:

```suggestion
      index += warps_per_grid;
```

Reply:
I guess I'm not understanding why I want to index like this.  Let's say for the sake of argument we have a grid size of 32 and 64 input elements.  2 warps, 64 threads.   

'index' here is into the input and output arrays (a thread index). If we increment that by warps_per_grid (2), tid = 0  would result in index = 0, 2.  Where what we want is index = 0, 32.

I artificially reduced the grid size as a test case and it definitely produces the wrong results.

Reply:
I added a test case (CopyIfElseTestTinyGrid) that specifically tests this case.  It sends in 64 element columns but launches the kernel as  kernel<<<1, 32>>> .  With 
```
index += block_size * gridDim.x 
```

it works correctly.  However, with 
```
index += warps_per_grid;
```

it does not.  The highest index will ever reach is 32 + loop count, since warps_per_grid = 1.

Reply:
Yep, you are correct.